Support dirhtml builder

jbms opened the pull request:
Fixes #110

2bndy5 commented:
Building this theme on windows is a pain (still need to use WSL). If I can download the artifacts, then it is just as good as doing the entire workflow locally (just to inspect the rendered output of a PR).

2bndy5 commented:
I suppose I could keep downloading the built wheel artifacts and build the docs myself using that, but it is just an extra step to getting a rendered output.

It is also handy to know that any future changes don't break this feature. The step for the latex builder has proven useful for that as well (and those builds are included in the artifacts).

2bndy5 commented:
Well I see you added Windows and Mac OS to the CI, but you can't use the wheels built from windows because the CSS and JS bundles aren't named correctly (see #18 - abridged hash is missing from the minified filenames). Thus the CI artifacts of the doc builds are... um... broken.

The doc build artifacts are being overwritten by the non-ubuntu jobs. Since that is currently the case, I hesitate to even install the wheel that is uploaded as an artifact.

2bndy5 commented on .github/workflows/build.yml:
```diff
@@ -8,6 +8,8 @@ jobs:
       matrix:
         os:
           - 'ubuntu-latest'
+          - 'windows-latest'
+          - 'macos-latest'
```
While I appreciate the inclusiveness here, the windows wheels are actually false-positive results (un-useable).

Personally, I'm rather anit-Apple, so I can't vouch for anything that VM produces. Github CI runners using macos actually have usage limits because they can't be served from Microsoft Azure servers. I doubt you'll hit these limits on this repo, but its yet another way Apple milks it's own enthusiasts for 💰.
```suggestion
          # - 'windows-latest'
          # - 'macos-latest'
```

2bndy5 commented:
The last real problem may actually be related to the html templates. This is the rendered result from a windows wheel:
```html
      <link rel="stylesheet" href="_static/stylesheets/main.css">
      
        
        <link rel="stylesheet" href="_static/stylesheets/palette.css">
        
```
At least the bundles' filenames are named correctly now.

jbms commented:
I think I have now fixed the issue of artifacts being overwritten, and I think I have also fixed #18 (sounds like maybe it is not quite fixed yet, though). We could file that as an upstream bug as well.

I agree that Apple makes it annoying to develop for macOS in many ways, but a lot of developers use macOS laptops so it is useful to support it if not too difficult.

2bndy5 commented:
Could rebase on latest main?

This is looking great to solve #110 !

2bndy5 commented on .github/workflows/build.yml:
```diff
@@ -48,45 +53,80 @@ jobs:
       run: sphinx-build . _build/html -W --keep-going -T
+    - name: Build dirhtml docs
+      working-directory: docs
+      run: sphinx-build . _build/dirhtml -W --keep-going -T
```
need to actually use the dirhtml builder here. I just realized I'm inspecting duplicates of the default html builder.
```suggestion
      run: sphinx-build . _build/dirhtml -W --keep-going -T -b dirhtml
```
